### DISCUSSION TITLE
Duplicate a Screen

### DISCUSSION THREAD

**Discussion by lxnunes** [Budibase/budibase#3768] [Ideas]
Hi!

I found myself creating several versions of the same screen, sometimes to be used as a regular edit screen, other times as a modal or even as a read-only screen. 

I know most (if not all) of these aspects can be controlled using Conditions, but sometimes it would be easier to just duplicate a screen for testing - just like duplicating screen objects. 

Is there a way to achieve this?

Best

**Comment by mjashanks** [comment]
HI @lxnunes - there is no way to duplicate a screen. However, you can copy and paste components - which works between screens.

![image](https://user-images.githubusercontent.com/3524181/145790626-af8b7658-cbde-4277-bfb8-ac0615c609aa.png)

So, you would:
1. Create a new screen.
2. Add any component to your screen.
2. Copy the outer-most component (which will include all the children), on the screen that you are trying to duplcate.
3. Goto your new screen and popup the context menu on your new component
4. Choose `Paste Below`.

The only issue with this right now, is that Bindings don't copy across correctly - so you will have to find anything that uses javascript or handlebars, and re-complete it.

If you have a lot of components, this should at least be quicker than recreating the entire screen manually.

However, I agree that duplicating a screen would be a useful feature. Turning this into an idea!

**Reply by lxnunes** [reply]
Hi @mjashanks, thank you for the suggestion and heads-up about the Bindings. 

One thought about Step nr 5, it only holds when there is a pre-existing component. The context menu for the Screen does not have such an option. Perhaps a "Paste inside" option could be added (and if a component already exists, added at the end)?

**Reply by mjashanks** [reply]
Yes - agreed. That's why I had the "Add component" step in (No.2). It's a bit of a hack!

**Reply by lxnunes** [reply]
Sorry, my bad, I think I just went straight to 5 after reading 1 :)
